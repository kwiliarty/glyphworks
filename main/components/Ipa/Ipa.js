import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.span`
  font-family: ${ props => props.theme.ipaFontFamily };
`

const Ipa = props => {
  return (
    <Wrapper>
      { props.children }
    </Wrapper>
  )
}

Ipa.propTypes = {
  /** This component renders its children. */
  children: PropTypes.node,
}

export default Ipa
