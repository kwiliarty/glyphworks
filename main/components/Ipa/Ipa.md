`Ipa` is a wrapper that we can put around IPA text.

```jsx
<p>
  We'll want to make sure that the line heights are even.
  You can insert IPA such as <Ipa>[ɡlɪfwə˞ks]</Ipa> into running text.
  We'll want to make sure that the line heights are even.
</p>
```
