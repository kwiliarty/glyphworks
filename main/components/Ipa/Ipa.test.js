import React from 'react'
import Ipa from './Ipa'
import { render, screen } from 'utils'

describe( 'An Ipa component', () => {
  it( 'displays its children in an IPA font', () => {
    render(
      <Ipa>test</Ipa>
    )
    const ipa = screen.getByText('test')
    expect( ipa ).toMatchSnapshot()
  })
})
