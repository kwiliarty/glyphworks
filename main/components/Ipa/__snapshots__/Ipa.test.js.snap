// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An Ipa component displays its children in an IPA font 1`] = `
.c0 {
  font-family: GentiumPlus,serif;
}

<span
  class="c0"
>
  test
</span>
`;
